src/twisted/web/server.py

```diff
@@ -522,6 +522,26 @@ def URLPath(self):
         return urlpath.URLPath.fromRequest(self)
 
 
+    def URL(self):
+        """
+        Return URL object of this request object.
+
+        @return: the URL
+        @rtype: L{hyperlink.DecodedURL}
+        """
+
+        # Pre process to decode from bytes data
+        _allascii = b"".join([chr(x).encode('ascii') for x in range(1, 128)])
+        quoted = quote(self.prePathURL(), safe=_allascii)
```

Comment:
So, this throws away the query string (and fragment) -- I don't think that's correct behavior, though that seems to be what URLPath currently does as well.

Reply:
Let's not repeat that mistake :)